nomt/src/beatree/ops/update/branch_updater.rs

```diff
@@ -34,15 +59,22 @@ impl BaseBranch {
     fn key_value(&self, i: usize) -> (Key, PageNumber) {
         (self.key(i), self.node.node_pointer(i).into())
     }
-
-    fn advance_iter(&mut self) {
-        self.iter_pos += 1;
-    }
 }
 
+// BranchOp used to create a new node starting off a possible base node.
+//
+// `Update` and `KeepChunk` refers to compressed separator present into
+// the base node. Every uncompressed separator present in the base or
+// that will be inserted in the new node will be represented as an `Insert` operation.
 enum BranchOp {
+    // Separator and its page number
     Insert(Key, PageNumber),
-    Keep(usize, usize),
+    // Contains the position at which the separator is saved in the base,
+    // along with the updated page number
+    Update(usize, PageNumber),
```

Comment:
LeafUpdater also has binary search but doesn't have the `Update`. 

I definitely see the motivation for this (swapping page numbers should be cheap, if the key doesn't change, and the separators can be completely copied with a normal `memcpy`).

Reply:
> LeafUpdater also has binary search but doesn't have the Update

That's totally right, and I think it could technically be added, but in that case, I don't think it would be worth it. The optimization takes place from the fact that the same `bitwise_memcpy` can be called on a bigger amount of bytes, assuming that the added computation is faster than calling `bitwise_memcpy` multiple times. I have the gut feeling that something like this could give a positive trade-off only for updated values that keep the same size. But this would require checks to make sure that the size is the same. Second, if the size changes, then a similar amount of byte shift would be required compared to what the current `Insert` and `KeepChunk` approach is doing.

Different from the branches, where the "values" are always 4-byte page numbers